### PR TITLE
Use macos12 machines in official build

### PR DESCRIPTION
**Pull request by agocke**
Scout out macos 12 images in official build only.

### CODE REVIEW

**Comment by agocke** [comment]
Also flighting this in PR first.

**Review by jkoritzinsky — APPROVED** [review]

**Review by hoyosjs — APPROVED** [review]

File: eng/pipelines/common/xplat-setup.yml [hunk 1]
```diff
@@ -146,10 +146,13 @@ jobs:
           name: $(DncEngInternalBuildPool)
           demands: ImageOverride -equals Build.Ubuntu.1804.Amd64
 
-        # OSX Build Pool (we don't have on-prem OSX BuildPool
-        ${{ if in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator') }}:
+        # OSX Build Pool (we don't have on-prem OSX BuildPool). Use 11 on public, 12 on internal.
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), eq(variables['System.TeamProject'], 'public')) }}:
           vmImage: 'macos-11'
 
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), ne(variables['System.TeamProject'], 'public')) }}:
```
Comment (hoyosjs) [inline eng/pipelines/common/xplat-setup.yml:153]: you can use `${{ else }}` here

Comment (agocke) [reply eng/pipelines/common/xplat-setup.yml:153]: Wait, is that new? That has to be new.

**Comment by agocke** [comment]
Official build passed, merging.

**Pull request merged**